src/Stack/Build/Execute.hs

```diff
@@ -593,6 +596,16 @@ singleBuild ac@ActionContext {..} ee@ExecuteEnv {..} task@Task {..} =
             TTLocal lp -> "build" : map T.unpack (Set.toList $ lpComponents lp)
             TTUpstream _ _ -> ["build"]
 
+    when (shouldBuildHaddock eeBuildOpts eeWanted (packageName package) &&
+          packageHasLibrary package &&
```

Comment:
Pretty sure it's redundant (I think I wrote this before I added the packageHasExposedModules check to shouldBuildHaddock).